### PR TITLE
chore(deps): update to latest base...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore(deps): update to latest base images

/cherry-pick 1.3.x

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: docker/Dockerfile
Comment (rm3l) [inline docker/Dockerfile:1]: Should we update [`.rhdh/docker/Dockerfile`](https://github.com/redhat-developer/rhdh-operator/blob/update-base-images/.rhdh/docker/Dockerfile) as well?

File: docker/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 #@follow_tag(registry.redhat.io/rhel9/go-toolset:latest)
-FROM registry.access.redhat.com/ubi9/go-toolset:1.21.11-7@sha256:d128c3d36878251f039606f144ef2608746c3203708b722295e6c571df1d8613 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21.11-7.1724233645@sha256:d128c3d36878251f039606f144ef2608746c3203708b722295e6c571df1d8613 AS builder
```
Comment (rm3l) [inline docker/Dockerfile:16]:
```suggestion
FROM registry.access.redhat.com/ubi9/go-toolset:1.21.11-9@sha256:ab6e4ea87075b71c67cae7c5f5131ce4508ac62d5c1b048781fed310573b4e95 AS builder
```

I can see a newer tag: `1.21.11-9`. Could we use that instead?
https://catalog.redhat.com/software/containers/ubi9/go-toolset/61e5c00b4ec9945c18787690?container-tabs=gti

File: docker/Dockerfile [hunk 2]
```diff
@@ -54,7 +54,7 @@ RUN export ARCH="$(uname -m)" && if [[ ${ARCH} == "x86_64" ]]; then export ARCH=
 
 # Install openssl for FIPS support
 #@follow_tag(registry.redhat.io/ubi9/ubi-minimal:latest)
-FROM registry.access.redhat.com/ubi9-minimal:9.4-1194@sha256:104cf11d890aeb7dd5728b7d7732e175a0e4018f1bb00d2faebcc8f6bf29bd52 AS runtime
+FROM registry.access.redhat.com/ubi9-minimal:9.4-1227@sha256:104cf11d890aeb7dd5728b7d7732e175a0e4018f1bb00d2faebcc8f6bf29bd52 AS runtime
```
Comment (rm3l) [inline docker/Dockerfile:57]:
```suggestion
FROM registry.access.redhat.com/ubi9-minimal:9.4-1227@sha256:35a12657ce1bcb2b7667f4e6e0147186c1e0172cc43ece5452ab85afd6532791 AS runtime
```
I can see a newer digest. Could we use that instead?
https://catalog.redhat.com/software/containers/ubi9-minimal/61832888c0d15aff4912fe0d?container-tabs=gti

File: docker/Dockerfile (hunk 1 above)
Comment (nickboldt) [reply docker/Dockerfile:16]: we definitely can. I wonder why this repo isn't getting renovate updates automatically?

**Comment by rm3l** [comment]
> we definitely can. I wonder why this repo isn't getting renovate updates automatically?

From the list of detected dependencies I can see in https://developer.mend.io/github/redhat-developer/rhdh-operator,  I guess it should. I noticed some Renovate updates for these images in https://github.com/redhat-developer/rhdh-operator/pull/153 for example (but not sure why there is a conflict - maybe the branch is outdated and needs to be rebased).

File: docker/Dockerfile
Comment (nickboldt) [reply docker/Dockerfile:1]: yes, technically. I was only updating the ones that affect the Konflux setup I'm trying to make work, but you're right.

**Pull request merged**